### PR TITLE
Market data getters

### LINKED ISSUE #72: SDK: Add charts graphql data call
- [x] Market price of Synth from DEX for range (yam-www white line on chart)
- [ ] Marketcap from lsp
- [x] TVL from api.yam.finance/tvl
- [x] DEX pool liquidity
- [x] pool volume last 24h

### PR DESCRIPTION
**Pull request by ghost**
### Description of changes proposed in this pull request:
- closes #72

### Checks:

- [x] Have you followed the guidelines in our Contributing document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### New Feature Submission:

- [x] Have you successfully ran tests with your changes locally?
- [x] Have you lint your code locally prior to submission?

### CODE REVIEW

**Comment by byterose** [comment]
@SnakePoison @ethedev can we merge this pr to avoid more conflicts?﻿

**Comment by byterose** [comment]
@ElmoArmy @ethedev can we merge this pr to avoid more conflicts?﻿﻿

**Comment by ethedev** [comment]
> @ElmoArmy @ethedev can we merge this pr to avoid more conflicts?﻿﻿
> 

merging shortly

File: test/SynthsSdk.test.ts [hunk 1]
```diff
@@ -49,9 +53,26 @@ describe("Synths SDKs", function () {
       upunksAsset = synthsSDK.connectAsset("upunks-0921");
     });
 
-    // @todo Add tests
+    // @todo Add tests.
     describe("Interact with asset", function () {
+      it("helpers - success", async function () {
+        this.timeout(100000);
+        console.log(await getYamSynthsTotalTVL());
```
Comment (ethedev) [inline test/SynthsSdk.test.ts:60]: add expect, with more details to what are we expecting here and for the console log you can keep it but comment it

File: test/SynthsSdk.test.ts [hunk 2]
```diff
@@ -49,9 +53,26 @@ describe("Synths SDKs", function () {
...
+
+        console.log(
+          await getSynthData(
+            "upunks-0921",
+            "0x86140A763077155964754968B6F6e243fE809cBe"
+          )
+        );
```
Comment (ethedev) [inline test/SynthsSdk.test.ts:67]: to separate, comment and add expect as well

File: test/SynthsSdk.test.ts [hunk 3]
```diff
@@ -49,9 +53,26 @@ describe("Synths SDKs", function () {
...
+
+        const monthAgoTimestamp = Math.floor(Date.now() / 1000) - 2629743;
+        console.log(
+          await getSynthChartData("0x86140A763077155964754968B6F6e243fE809cBe")
+        );
+      });
```
Comment (ethedev) [inline test/SynthsSdk.test.ts:73]: same

File: test/SynthsSdk.test.ts [hunk 4]
```diff
@@ -49,9 +53,26 @@ describe("Synths SDKs", function () {
...
+        console.log(await getYamSynthsTotalTVL());
+
+        console.log(
+          await getSynthData(
+            "upunks-0921",
+            "0x86140A763077155964754968B6F6e243fE809cBe"
+          )
+        );
+
+        const monthAgoTimestamp = Math.floor(Date.now() / 1000) - 2629743;
```
Comment (ethedev) [inline test/SynthsSdk.test.ts:69]: whats this for?

**Review by ethedev — CHANGES_REQUESTED** [review]

**Comment by byterose** [comment]
Tests need to be updated in a separate pr. This was just for logging - which I removed for now.
The interface will be changed and then I will create more detailed tests.

**Comment by ethedev** [comment]
> Tests need to be updated in a separate pr. This was just for logging - which I removed for now. The interface will be changed and then I will create more detailed tests.

everything related should be in this pr as well as tests without any console logs unless specified, then for the interface changes it can still be changed later and once that happen you can update tests accordingly

**Comment by byterose** [comment]
@ethedev I updated tests as requested.

File: test/SynthsSdk.test.ts [hunk 5]
```diff
@@ -49,9 +54,36 @@ describe("Synths SDKs", function () {
...
+          `https://data.yam.finance/degenerative/apr/upunks-0921`
+        );
+
+        expect(synthData).to.deep.include({
+          // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+          apr: response.data["aprMultiplier"] as string,
+        });
+        expect(synthChartData).to.be.an("array");
+        expect(synthPrice).to.not.equal(0);
+      });
```
Comment (ethedev) [inline test/SynthsSdk.test.ts:84]: for now thats ok but as a next step keep in mind to separate all the helper functions each individually, to eventually become something like that:
- ...
- Interact with asset
  - Asset helpers
    - `getCurrentDexTokenPrice - success`
    - `getSynthData- success`
    - `getSynthChartData- success`
    - etc.
- ...

**Review by ethedev — COMMENTED** [review]
we need to delete this file `src/lib/.Asset.ts.swp` and gitignore `*.swp` files if it keeps coming

File: test/SynthsSdk.test.ts (hunk 5 above)
Comment (byterose) [reply test/SynthsSdk.test.ts:84]: Yes, I will update these  tests soon in another pr.

**Review by ethedev — APPROVED** [review]

**Pull request merged**